macros/light-sdk-macros/tests/hasher.rs

```diff
@@ -28,51 +29,392 @@ pub struct MyNestedNonHashableStruct {
+                data: String,
+            }
+
+            let large_data = "a".repeat(64);
+            let truncated = LongTruncatedStruct {
+                data: large_data.clone(),
+            };
+
+            let mut modified_data = large_data.clone();
+            modified_data.push('b');
+            let truncated2 = LongTruncatedStruct {
+                data: modified_data,
+            };
+
+            let hash1 = truncated.hash::<Poseidon>().unwrap();
+            let hash2 = truncated2.hash::<Poseidon>().unwrap();
+
+            assert_ne!(hash1, hash2);
+        }
+
+        #[test]
+        fn test_multiple_truncates() {
+            #[derive(LightHasher)]
+            struct MultiTruncate {
+                #[truncate]
+                data1: String,
+                #[truncate]
+                data2: String,
+            }
+
+            let test_struct = MultiTruncate {
+                data1: "a".repeat(64),
+                data2: "b".repeat(64),
+            };
+
+            let hash1 = test_struct.hash::<Poseidon>().unwrap();
+
+            // Modifying data after truncation point shouldn't affect hash
+            let test_struct2 = MultiTruncate {
+                data1: "a".repeat(65),
+                data2: "b".repeat(65),
+            };
+
+            let hash2 = test_struct2.hash::<Poseidon>().unwrap();
+            assert_ne!(
+                hash1, hash2,
+                "Different data should produce different hashes"
+            );
+        }
+
+        #[test]
+        fn test_nested_with_truncate() {
+            #[derive(LightHasher)]
+            struct NestedTruncate {
+                #[nested]
+                inner: MyNestedStruct,
+                #[truncate]
+                data: String,
+            }
+
+            let nested = create_test_nested_struct();
+            let test_struct = NestedTruncate {
+                inner: nested,
+                data: "test".to_string(),
+            };
+
+            let hash = test_struct.hash::<Poseidon>().unwrap();
+            assert_eq!(hash.len(), 32);
+        }
+    }
+
+    mod nesting_tests {
+        use super::*;
+
+        #[test]
+        fn test_recursive_nesting() {
+            let nested_struct = create_test_nested_struct();
+
+            #[derive(LightHasher)]
+            struct TestNestedStruct {
+                #[nested]
+                one: MyNestedStruct,
+                #[nested]
+                two: MyNestedStruct,
+            }
+
+            let test_nested_struct = TestNestedStruct {
+                one: nested_struct,
+                two: create_test_nested_struct(),
+            };
+
+            let manual_hash = Poseidon::hashv(&[
+                &test_nested_struct.one.hash::<Poseidon>().unwrap(),
+                &test_nested_struct.two.hash::<Poseidon>().unwrap(),
+            ])
+            .unwrap();
+
+            assert_eq!(test_nested_struct.hash::<Poseidon>().unwrap(), manual_hash);
```

Comment:
can you add a hardcoded reference value for this and the tests that you changed?
Can you add a truncated value inside `MyNestedStruct`?